README.md

```diff
@@ -51,6 +51,29 @@ git pull && git annex get .
 
 See more at [the official documentation](https://git-annex.branchable.com/walkthrough/) and take note of our [in-lab troubleshooting](https://github.com/neuropoly/data-management/blob/master/git-annex.md).
 
+## Working from a forked repository
+
+> ⚠️ For advanced users only. Normally the instructions under [Download](#Download) should be enough.
+
+If you have forked this repository on Github (so that you have a copy `your-user-name/data-multi-subject` of `spine-generic/data-multi-subject`), you will need to take a few extra synchronization steps to get the latest data with `git annex get`.
+
+1. In your local clone of `your-user-name/data-multi-subject`, make sure that `spine-generic/data-multi-subject` is also configured as a remote:
+   ```
+   git remote -v
+   # the answer should show both your-user-name/data-multi-subject.git (probably named "origin")
+   # and spine-generic/data-multi-subject (probably named "upstream")
+   ```
+
+   If `spine-generic/data-multi-subject` is missing, you can add it with:
+   ```
+   git remote add upstream https://github.com/spine-generic/data-multi-subject.git
+   ```
+
+2. Then, to update your local clone, make sure to fetch the `git-annex` branch from `spine-generic/data-multi-subject` before running `git annex get`:
+   ```
+   git fetch upstream && git pull && git annex get .
+   ```
```

Comment:
I think `annex-readonly` is the expected way to handle this use case.

````suggestion
   If `spine-generic/data-multi-subject` is missing, you can add it with:
   ```
   git remote add upstream https://github.com/spine-generic/data-multi-subject.git
   git config remote.upstream.annex-readonly true
   ```

2. Then, to update your local clone, make sure to synchronize the `git-annex` branches with `spine-generic/data-multi-subject` before running `git annex get`:
   ```
   git annex sync -a && git annex get .
   ```
````

But I don't really trust git-annex's features to actually simplify things without adding a bunch of extra caveats.

I'm going to test both out and get back to you.

Reply:
## Test Environment

I've made https://github.com/neuropoly/data-multi-subject-test and forked it to https://github.com/kousu/data-multi-subject-test.

I added a commit: https://github.com/neuropoly/data-multi-subject-test/commit/11a1202ad00f9e8459d74bbc86aec3b7769497bd

<details>

```
p115628@joplin:~/datasets$ git clone git@github.com:neuropoly/data-multi-subject-test.git
Cloning into 'data-multi-subject-test'...
remote: Enumerating objects: 114767, done.
remote: Counting objects: 100% (18219/18219), done.
remote: Compressing objects: 100% (13988/13988), done.
remote: Total 114767 (delta 3337), reused 17233 (delta 3299), pack-reused 96548
Receiving objects: 100% (114767/114767), 14.13 MiB | 15.60 MiB/s, done.
Resolving deltas: 100% (43538/43538), done.
p115628@joplin:~/datasets$ cd data-multi-subject-test/
p115628@joplin:~/datasets/data-multi-subject-test$ git annex get sub-amu01
(merging origin/git-annex into git-annex...)
(recording state in git...)
(scanning for unlocked files...)
get sub-amu01/anat/sub-amu01_T1w.nii.gz (from amazon...) 
(checksum...) ok                      
get sub-amu01/anat/sub-amu01_T2star.nii.gz (from amazon...) 
(checksum...) ok                   
get sub-amu01/anat/sub-amu01_T2w.nii.gz (from amazon...) 
(checksum...) ok                   
get sub-amu01/anat/sub-amu01_acq-MToff_MTS.nii.gz (from amazon...) 
(checksum...) ok                   
get sub-amu01/anat/sub-amu01_acq-MTon_MTS.nii.gz (from amazon...) 
(checksum...) ok                   
get sub-amu01/anat/sub-amu01_acq-T1w_MTS.nii.gz (from amazon...) 
(checksum...) ok                  
get sub-amu01/dwi/sub-amu01_acq-b0_dwi.nii.gz (from amazon...) 
(checksum...) ok                   
get sub-amu01/dwi/sub-amu01_dwi.nii.gz (from amazon...) 
(checksum...) ok                     
(recording state in git...)
p115628@joplin:~/datasets/data-multi-subject-test$ dd if=/dev/urandom of=bomp.nii.gz count=50 bs=1M
50+0 records in
50+0 records out
52428800 bytes (52 MB, 50 MiB) copied, 0.33278 s, 158 MB/s
p115628@joplin:~/datasets/data-multi-subject-test$ GIT_TRACE=1 git add bomp.nii.gz 
01:51:28.080590 git.c:455               trace: built-in: git add bomp.nii.gz
01:51:28.086529 run-command.c:668       trace: run_command: 'git-annex smudge --clean -- '\''bomp.nii.gz'\'''
01:51:28.096328 git.c:455               trace: built-in: git config --null --list
01:51:28.101261 git.c:455               trace: built-in: git cat-file --batch
01:51:28.101527 git.c:455               trace: built-in: git cat-file '--batch-check=%(objectname) %(objecttype) %(objectsize)'
01:51:28.106885 git.c:455               trace: built-in: git show-ref git-annex
01:51:28.110612 git.c:455               trace: built-in: git show-ref --hash refs/heads/git-annex
01:51:28.114499 git.c:455               trace: built-in: git log refs/heads/git-annex..2f97732bab92f6e24ef7b9bb2b4f6b82bb286889 '--pretty=%H' -n1
01:51:28.120811 git.c:455               trace: built-in: git cat-file --batch
01:51:28.121380 git.c:455               trace: built-in: git cat-file '--batch-check=%(objectname) %(objecttype) %(objectsize)'
01:51:28.125595 git.c:455               trace: built-in: git check-attr -z --stdin annex.backend annex.largefiles annex.numcopies annex.mincopies --
01:51:28.506242 run-command.c:668       trace: run_command: 'git-annex smudge --clean -- '\''sub-amu01/dwi/sub-amu01_dwi.nii.gz'\'''
01:51:28.514841 git.c:455               trace: built-in: git config --null --list
01:51:28.519155 git.c:455               trace: built-in: git cat-file --batch
01:51:28.519538 git.c:455               trace: built-in: git cat-file '--batch-check=%(objectname) %(objecttype) %(objectsize)'
01:51:28.527198 git.c:455               trace: built-in: git show-ref git-annex
01:51:28.530271 git.c:455               trace: built-in: git show-ref --hash refs/heads/git-annex
01:51:28.533756 git.c:455               trace: built-in: git log refs/heads/git-annex..2f97732bab92f6e24ef7b9bb2b4f6b82bb286889 '--pretty=%H' -n1
01:51:28.539487 git.c:455               trace: built-in: git hash-object -w --stdin-paths --no-filters
01:51:28.539747 git.c:455               trace: built-in: git update-index -z --index-info
01:51:28.559863 git.c:455               trace: built-in: git cat-file --batch
01:51:28.561230 git.c:455               trace: built-in: git cat-file '--batch-check=%(objectname) %(objecttype) %(objectsize)'
01:51:28.565213 git.c:455               trace: built-in: git write-tree
01:51:28.601533 git.c:455               trace: built-in: git commit-tree 55ad8fcb8c57d8c82329250cd5fb2771449574fc --no-gpg-sign -p refs/heads/git-annex
01:51:28.604879 git.c:455               trace: built-in: git update-ref refs/heads/git-annex b36633716415396ab383f932ab95009893225bfb
01:51:28.610901 git.c:455               trace: built-in: git check-attr -z --stdin annex.backend annex.largefiles annex.numcopies annex.mincopies --
p115628@joplin:~/datasets/data-multi-subject-test$ git commit -m "Make an update"
[master 11a1202ad] Make an update
 1 file changed, 1 insertion(+)
 create mode 100644 bomp.nii.gz
p115628@joplin:~/datasets/data-multi-subject-test$ git log -p
commit 11a1202ad00f9e8459d74bbc86aec3b7769497bd (HEAD -> master)
Author: Nick Guenther <nick.guenther@polymtl.ca>
Date:   Wed Aug 24 01:51:39 2022 -0400

    Make an update

diff --git a/bomp.nii.gz b/bomp.nii.gz
new file mode 100644
index 000000000..ac9f94e6e
--- /dev/null
+++ b/bomp.nii.gz
@@ -0,0 +1 @@
+/annex/objects/SHA256E-s52428800--0464317b90204cd82ec7ab4ce1f9ec8d2feb875f3e510e786054c52f8485fe78.nii.gz

commit 42e58865e3f26266d6b9899518942e317c85721f (origin/master, origin/HEAD)
Author: mbondy023 <98548430+mbondy023@users.noreply.github.com>
Date:   Wed Jul 20 14:12:08 2022 -0500

    add brain_t1 category into exclude.yml file (#127)

diff --git a/exclude.yml b/exclude.yml
index 35eab866e..063eabdee 100644
--- a/exclude.yml
+++ b/exclude.yml
@@ -55,3 +55,26 @@ dti_rd:
 - sub-mountSinai03  # T2w was re-acquired (hence wrong T2w -> DWI registration)
 - sub-oxfordFmrib04  # T1w scan is not aligned with other contrasts (subject repositioning)
 - sub-oxfordFmrib01  # registration issue (segmentation OK)
+brain_t1:
+- sub-barcelona01  # wrong FOV placement
+- sub-barcelona02  # wrong FOV placement
+- sub-barcelona03  # wrong FOV placement
+- sub-barcelona04  # wrong FOV placement
p115628@joplin:~/datasets/data-multi-subject-test$ git annex copy --to=amazon
copy bomp.nii.gz (checking amazon...) (to amazon...) 
16%   7.93 MiB         21 MiB/s 2s
  S3Error {s3StatusCode = Status {statusCode = 400, statusMessage = "Bad Request"}, s3ErrorCode = "AccessControlListNotSupported", s3ErrorMessage = "The bucket does not allow ACLs", s3ErrorResource = Nothing, s3ErrorHostId = Just "Px4NqkAtphKYzJhL+Z63XdxPaZL5g8fnJkrqgz6G8dA279RiEFIQvfybtMm62tAQRcJAOz6yeR0=", s3ErrorAccessKeyId = Nothing, s3ErrorStringToSign = Nothing, s3ErrorBucket = Nothing, s3ErrorEndpointRaw = Nothing, s3ErrorEndpoint = Nothing}

16%   7.96 MiB         20 MiB/s 2s 
  S3Error {s3StatusCode = Status {statusCode = 400, statusMessage = "Bad Request"}, s3ErrorCode = "AccessControlListNotSupported", s3ErrorMessage = "The bucket does not allow ACLs", s3ErrorResource = Nothing, s3ErrorHostId = Just "r8IoV2S8mo9LOgfIhcm2vpR7VfX2kmuiDnn555GJZysVDZzDKswY8G9k7swhD0PrLokvWc4Th7o=", s3ErrorAccessKeyId = Nothing, s3ErrorStringToSign = Nothing, s3ErrorBucket = Nothing, s3ErrorEndpointRaw = Nothing, s3ErrorEndpoint = Nothing}

16%   7.96 MiB         20 MiB/s 2s 
  S3Error {s3StatusCode = Status {statusCode = 400, statusMessage = "Bad Request"}, s3ErrorCode = "AccessControlListNotSupported", s3ErrorMessage = "The bucket does not allow ACLs", s3ErrorResource = Nothing, s3ErrorHostId = Just "c98qZp+RhR0E2ygTWS8HJAivLoTydwpRwHib7eJPFU+lnl1hv4AibrQeBTQmy7YvxzG64Jg4Tmo=", s3ErrorAccessKeyId = Nothing, s3ErrorStringToSign = Nothing, s3ErrorBucket = Nothing, s3ErrorEndpointRaw = Nothing, s3ErrorEndpoint = Nothing}
failed
copy sub-amu01/anat/sub-amu01_T1w.nii.gz (checking amazon...) ok
copy sub-amu01/anat/sub-amu01_T2star.nii.gz (checking amazon...) ok
copy sub-amu01/anat/sub-amu01_T2w.nii.gz (checking amazon...) ok
copy sub-amu01/anat/sub-amu01_acq-MToff_MTS.nii.gz (checking amazon...) ok
copy sub-amu01/anat/sub-amu01_acq-MTon_MTS.nii.gz (checking amazon...) ok
copy sub-amu01/anat/sub-amu01_acq-T1w_MTS.nii.gz (checking amazon...) ok
copy sub-amu01/dwi/sub-amu01_acq-b0_dwi.nii.gz (checking amazon...) ok
copy sub-amu01/dwi/sub-amu01_dwi.nii.gz (checking amazon...) ok
git-annex: copy: 1 failed
p115628@joplin:~/datasets/data-multi-subject-test$ git annex enableremote amazon public=no   # enable write access just for you
enableremote amazon ok
(recording state in git...)
p115628@joplin:~/datasets/data-multi-subject-test$ git annex copy --to amazon                # upload data to the amazon mirror
copy bomp.nii.gz (checking amazon...) (to amazon...) 
ok                                 
copy sub-amu01/anat/sub-amu01_T1w.nii.gz (checking amazon...) ok
copy sub-amu01/anat/sub-amu01_T2star.nii.gz (checking amazon...) ok
copy sub-amu01/anat/sub-amu01_T2w.nii.gz (checking amazon...) ok
copy sub-amu01/anat/sub-amu01_acq-MToff_MTS.nii.gz (checking amazon...) ok
copy sub-amu01/anat/sub-amu01_acq-MTon_MTS.nii.gz (checking amazon...) ok
copy sub-amu01/anat/sub-amu01_acq-T1w_MTS.nii.gz (checking amazon...) ok
copy sub-amu01/dwi/sub-amu01_acq-b0_dwi.nii.gz (checking amazon...) ok
copy sub-amu01/dwi/sub-amu01_dwi.nii.gz (checking amazon...) ok
(recording state in git...)
p115628@joplin:~/datasets/data-multi-subject-test$ git annex enableremote amazon public=yes  # re-enable read access for everyone else
enableremote amazon ok
(recording state in git...)
p115628@joplin:~/datasets/data-multi-subject-test$ git annex sync --no-content --only-annex  # upload annex metadata
pull origin 
fatal: couldn't find remote ref refs/heads/synced/git-annex
ok
push origin 
Enumerating objects: 76, done.
Counting objects: 100% (76/76), done.
Delta compression using up to 128 threads
Compressing objects: 100% (43/43), done.
Writing objects: 100% (48/48), 4.39 KiB | 321.00 KiB/s, done.
Total 48 (delta 28), reused 1 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (28/28), completed with 21 local objects.
remote: 
remote: Create a pull request for 'synced/git-annex' on GitHub by visiting:
remote:      https://github.com/neuropoly/data-multi-subject-test/pull/new/synced/git-annex
remote: 
To github.com:neuropoly/data-multi-subject-test.git
 * [new branch]          git-annex -> synced/git-annex
ok
p115628@joplin:~/datasets/data-multi-subject-test$ git push                                  # upload your branch
Enumerating objects: 4, done.
Counting objects: 100% (4/4), done.
Delta compression using up to 128 threads
Compressing objects: 100% (3/3), done.
Writing objects: 100% (3/3), 381 bytes | 381.00 KiB/s, done.
Total 3 (delta 1), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (1/1), completed with 1 local object.
To github.com:neuropoly/data-multi-subject-test.git
   42e58865e..11a1202ad  master -> master
```

</details>

now the @kousu copy is out of sync.

I switched to working on the fork:

<details>

```
p115628@joplin:~/datasets$ chmod -R +w data-multi-subject-test/
p115628@joplin:~/datasets$ git clone git@github.com:kousu/data-multi-subject-test.git
fatal: destination path 'data-multi-subject-test' already exists and is not an empty directory.
p115628@joplin:~/datasets$ rm -rf data-multi-subject-test/
p115628@joplin:~/datasets$ git clone git@github.com:kousu/data-multi-subject-test.git
Cloning into 'data-multi-subject-test'...
remote: Enumerating objects: 114767, done.
remote: Counting objects: 100% (18219/18219), done.
remote: Compressing objects: 100% (13988/13988), done.
remote: Total 114767 (delta 3337), reused 17233 (delta 3299), pack-reused 96548
Receiving objects: 100% (114767/114767), 14.13 MiB | 14.61 MiB/s, done.
Resolving deltas: 100% (43538/43538), done.
p115628@joplin:~/datasets$ cd data-multi-subject-test/
p115628@joplin:~/datasets/data-multi-subject-test$ git remote -v
origin  git@github.com:kousu/data-multi-subject-test.git (fetch)
origin  git@github.com:kousu/data-multi-subject-test.git (push)
```

</details>

Then, as @renelabounek [did](https://github.com/spine-generic/data-multi-subject/issues/125#issue-1305234610), I used the GitHub UI to sync up the fork:

![Screenshot 2022-08-24 at 01-58-38 kousu_data-multi-subject-test Multi-subject data for the Spine Generic project](https://user-images.githubusercontent.com/987487/186341802-be98ca15-e7b1-4215-a52e-e0579fcfc937.png)
![Screenshot 2022-08-24 at 01-58-52 kousu_data-multi-subject-test Multi-subject data for the Spine Generic project](https://user-images.githubusercontent.com/987487/186341807-79da8c2c-75c5-4c77-9b57-678dd64f4360.png)

Then I did

```
git pull && git annex get bomp.nii.gz
```

And hit the failure:

```
p115628@joplin:~/datasets/data-multi-subject-test$ git pull && git annex get bomp.nii.gz
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 3 (delta 1), reused 3 (delta 1), pack-reused 0
Unpacking objects: 100% (3/3), 361 bytes | 180.00 KiB/s, done.
From github.com:kousu/data-multi-subject-test
   42e58865e..11a1202ad  master     -> origin/master
Updating 42e58865e..11a1202ad
Fast-forward
 bomp.nii.gz | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 bomp.nii.gz
(merging origin/git-annex into git-annex...)
(recording state in git...)
(scanning for unlocked files...)
get bomp.nii.gz (not available) 
  No other repository is known to contain the file.

  (Note that these git remotes have annex-ignore set: origin)
failed
git-annex: get: 1 failed
```

## Your fix

Now I'm going to try your fix:

<details>

```
p115628@joplin:~/datasets/data-multi-subject-test$ git remote add upstream https://github.com/neuropoly/data-multi-subject-test.git
p115628@joplin:~/datasets/data-multi-subject-test$  git fetch upstream && git pull && git annex get bomp.nii.gz
remote: Enumerating objects: 47, done.
remote: Counting objects: 100% (47/47), done.
remote: Compressing objects: 100% (15/15), done.
remote: Total 47 (delta 28), reused 47 (delta 28), pack-reused 0
Unpacking objects: 100% (47/47), 4.36 KiB | 99.00 KiB/s, done.
From https://github.com/neuropoly/data-multi-subject-test
 * [new branch]          af/fix_images                        -> upstream/af/fix_images
 * [new branch]          af/update_participants_tsv           -> upstream/af/update_participants_tsv
 * [new branch]          af/update_sub-ucl06_GH_direct        -> upstream/af/update_sub-ucl06_GH_direct
 * [new branch]          git-annex                            -> upstream/git-annex
 * [new branch]          master                               -> upstream/master
 * [new branch]          mguaypaq/work-from-fork              -> upstream/mguaypaq/work-from-fork
 * [new branch]          mt-entity                            -> upstream/mt-entity
 * [new branch]          ng/ci-cache                          -> upstream/ng/ci-cache
 * [new branch]          renelabounek-master                  -> upstream/renelabounek-master
 * [new branch]          rl/height-weight                     -> upstream/rl/height-weight
 * [new branch]          sb/121-move-derivatives              -> upstream/sb/121-move-derivatives
 * [new branch]          synced/af/csfseg_json                -> upstream/synced/af/csfseg_json
 * [new branch]          synced/git-annex                     -> upstream/synced/git-annex
 * [new branch]          synced/jca/17-defacing               -> upstream/synced/jca/17-defacing
 * [new branch]          synced/jca/21-travis                 -> upstream/synced/jca/21-travis
 * [new branch]          synced/jca/7-check-data-consistency  -> upstream/synced/jca/7-check-data-consistency
 * [new branch]          synced/jca/derivatives-r20200830-mts -> upstream/synced/jca/derivatives-r20200830-mts
 * [new branch]          synced/jca/participants              -> upstream/synced/jca/participants
 * [new branch]          synced/jca/readme                    -> upstream/synced/jca/readme
 * [new branch]          synced/jca/reupload-disc-labels      -> upstream/synced/jca/reupload-disc-labels
 * [new branch]          synced/jca/update-derivatives        -> upstream/synced/jca/update-derivatives
Already up to date.
get bomp.nii.gz (merging upstream/git-annex into git-annex...)
(recording state in git...)

  Remote upstream not usable by git-annex; setting annex-ignore
(from amazon...) 
(checksum...) ok                     
(recording state in git...)
p115628@joplin:~/datasets/data-multi-subject-test$ file bomp.nii.gz 
bomp.nii.gz: data
p115628@joplin:~/datasets/data-multi-subject-test$ ls -lh bomp.nii.gz 
-rw------- 2 p115628 domain users 50M Aug 24 02:09 bomp.nii.gz
```

</details>

:heavy_check_mark: 


## My fix


<details><summary>Fresh slate</summary>

```
p115628@joplin:~/datasets/data-multi-subject-test$ cd ..
p115628@joplin:~/datasets$ rm -rf data-multi-subject-test/
rm: cannot remove 'data-multi-subject-test/.git/annex/objects/jw/Gf/SHA256E-s52428800--0464317b90204cd82ec7ab4ce1f9ec8d2feb875f3e510e786054c52f8485fe78.nii.gz/SHA256E-s52428800--0464317b90204cd82ec7ab4ce1f9ec8d2feb875f3e510e786054c52f8485fe78.nii.gz': Permission denied
p115628@joplin:~/datasets$ chmod -R +w data-multi-subject-test/
p115628@joplin:~/datasets$ rm -rf data-multi-subject-test/
p115628@joplin:~/datasets$ git clone git@github.com:kousu/data-multi-subject-test.git
Cloning into 'data-multi-subject-test'...
remote: Enumerating objects: 114770, done.
remote: Counting objects: 100% (18213/18213), done.
remote: Compressing objects: 100% (13988/13988), done.
remote: Total 114770 (delta 3330), reused 17229 (delta 3293), pack-reused 96557
Receiving objects: 100% (114770/114770), 14.12 MiB | 12.54 MiB/s, done.
Resolving deltas: 100% (43543/43543), done.
p115628@joplin:~/datasets$ cd data-multi-subject-test
p115628@joplin:~/datasets/data-multi-subject-test$ 
```

</details>

Then:

```
p115628@joplin:~/datasets/data-multi-subject-test$ git remote add upstream https://github.com/neuropoly/data-multi-subject-test.git
p115628@joplin:~/datasets/data-multi-subject-test$ git config remote.upstream.annex-readonly true
p115628@joplin:~/datasets/data-multi-subject-test$ git annex sync -a && git annex get bomp.nii.gz
(merging origin/git-annex into git-annex...)
(recording state in git...)
(scanning for unlocked files...)
pull upstream 
remote: Enumerating objects: 76, done.
remote: Counting objects: 100% (76/76), done.
remote: Compressing objects: 100% (15/15), done.
remote: Total 48 (delta 28), reused 48 (delta 28), pack-reused 0
Unpacking objects: 100% (48/48), 4.37 KiB | 97.00 KiB/s, done.
From https://github.com/neuropoly/data-multi-subject-test
 * branch                git-annex        -> FETCH_HEAD
 * branch                synced/git-annex -> FETCH_HEAD
 * [new branch]          git-annex        -> upstream/git-annex
 * [new branch]          synced/git-annex -> upstream/synced/git-annex
ok
pull origin 
From github.com:kousu/data-multi-subject-test
 * branch                git-annex  -> FETCH_HEAD
 * branch                synced/git-annex -> FETCH_HEAD
ok
(merging upstream/git-annex into git-annex...)
(recording state in git...)
get bomp.nii.gz (from amazon...) 
(checksum...) ok                      
get derivatives/labels/sub-amu01/anat/sub-amu01_T1w_labels-disc-manual.nii.gz (from amazon...) 
(checksum...) ok                  
get derivatives/labels/sub-amu01/anat/sub-amu01_T2w_labels-disc-manual.nii.gz (from amazon...) 
(checksum...) ok                  
get derivatives/labels/sub-amu02/anat/sub-amu02_T1w_RPI_r_labels-manual.nii.gz (from amazon...) 
(checksum...) ok                  
get derivatives/labels/sub-amu02/anat/sub-amu02_T1w_labels-disc-manual.nii.gz (from amazon...) 
(checksum...) ok                  
get derivatives/labels/sub-amu02/anat/sub-amu02_T2w_labels-disc-manual.nii.gz (from amazon...) 
(checksum...) ok                  
get derivatives/labels/sub-amu03/anat/sub-amu03_T1w_labels-disc-manual.nii.gz (from amazon...) 
(checksum...) ok                  
get derivatives/labels/sub-amu03/anat/sub-amu03_T2w_labels-disc-manual.nii.gz (from amazon...) 
(checksum...) ok                  
get derivatives/labels/sub-amu04/anat/sub-amu04_T1w_RPI_r_labels-manual.nii.gz (from amazon...) 
(checksum...) ok                  
get derivatives/labels/sub-amu04/anat/sub-amu04_T1w_labels-disc-manual.nii.gz (from amazon...) 
(checksum...) ok                  
get derivatives/labels/sub-amu04/anat/sub-amu04_T2w_labels-disc-manual.nii.gz (from amazon...) 
(checksum...) ok                  
get derivatives/labels/sub-amu05/anat/sub-amu05_T1w_RPI_r_labels-manual.nii.gz (from amazon...) 
3%    16 KiB          373 KiB/s 1s^C
```

Ack, `sync` did `sync --content`. Nooo.

:x: 

## My Fix, Redux

````suggestion
   If `spine-generic/data-multi-subject` is missing, you can add it with:
   ```
   git remote add upstream https://github.com/spine-generic/data-multi-subject.git
   git config remote.upstream.annex-readonly true
   ```
2. Then, to update your local clone, make sure to synchronize the `git-annex` branches with `spine-generic/data-multi-subject` before running `git annex get`:
   ```
   git annex sync -a --no-content && git pull && git annex get .
   ```
````

```
p115628@joplin:~/datasets$ git clone git@github.com:kousu/data-multi-subject-test.git
Cloning into 'data-multi-subject-test'...
remote: Enumerating objects: 125579, done.
remote: Counting objects: 100% (29031/29031), done.
remote: Compressing objects: 100% (21176/21176), done.
remote: Total 125579 (delta 6420), reused 27991 (delta 6359), pack-reused 96548
Receiving objects: 100% (125579/125579), 15.77 MiB | 16.04 MiB/s, done.
Resolving deltas: 100% (46621/46621), done.
p115628@joplin:~/datasets$ cd data-multi-subject-test/
p115628@joplin:~/datasets/data-multi-subject-test$ git ^C
p115628@joplin:~/datasets/data-multi-subject-test$ git remote add upstream https://github.com/neuropoly/data-multi-subject-test.git
p115628@joplin:~/datasets/data-multi-subject-test$ git config remote.upstream.annex-readonly true
p115628@joplin:~/datasets/data-multi-subject-test$ git annex sync -a --no-content && git pull && git annex get bomp.nii.gz 
(merging origin/git-annex into git-annex...)
(recording state in git...)
(scanning for unlocked files...)
pull upstream 
From https://github.com/neuropoly/data-multi-subject-test
 * branch                git-annex        -> FETCH_HEAD
 * branch                synced/git-annex -> FETCH_HEAD
 * [new branch]          git-annex        -> upstream/git-annex
 * [new branch]          synced/git-annex -> upstream/synced/git-annex
ok
pull origin 
From github.com:kousu/data-multi-subject-test
 * branch                git-annex  -> FETCH_HEAD
 * branch                synced/git-annex -> FETCH_HEAD
ok
push origin 
Enumerating objects: 8, done.
Counting objects: 100% (8/8), done.
Delta compression using up to 128 threads
Compressing objects: 100% (5/5), done.
Writing objects: 100% (6/6), 639 bytes | 58.00 KiB/s, done.
Total 6 (delta 3), reused 1 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (3/3), completed with 2 local objects.
To github.com:kousu/data-multi-subject-test.git
   f8985b72c..a61529b66  git-annex -> synced/git-annex
ok
Already up to date.
get bomp.nii.gz (from amazon...) 
(checksum...) ok                      
(recording state in git...)
```


:heavy_check_mark:

Reply:
Since the problem really comes when mixing GUI and CLI syncs, maybe we can fix it by doing all the syncing in the GUI or all the syncing in the CLI.

## GUI

### Test Setup

<details>

```
p115628@joplin:~/data-multi-subject-test$ git remote -v
amazon
origin  git@github.com:neuropoly/data-multi-subject-test.git (fetch)
origin  git@github.com:neuropoly/data-multi-subject-test.git (push)
p115628@joplin:~/data-multi-subject-test$ dd if=/dev/urandom of=bomp.nii.gz bs=1M count=3
3+0 records in
3+0 records out
3145728 bytes (3.1 MB, 3.0 MiB) copied, 0.0275007 s, 114 MB/s
p115628@joplin:~/data-multi-subject-test$ git add bomp.nii.gz 
p115628@joplin:~/data-multi-subject-test$ git commit -m "Update bomp"
[master 77e80fcbd] Update bomp
 1 file changed, 1 insertion(+), 1 deletion(-)
p115628@joplin:~/data-multi-subject-test$ git config annex.private true
p115628@joplin:~/data-multi-subject-test$ git annex enableremote amazon public=no   # enable write access just for you
enableremote amazon ok
(recording state in git...)
p115628@joplin:~/data-multi-subject-test$ git annex copy --to amazon                # upload data to the amazon mirror
copy bomp.nii.gz (checking amazon...) (to amazon...) 
ok                                
(recording state in git...)
p115628@joplin:~/data-multi-subject-test$ git annex enableremote amazon public=yes  # re-enable read access for everyone else
enableremote amazon ok
(recording state in git...)
p115628@joplin:~/data-multi-subject-test$ git annex sync --no-content --only-annex  # upload annex metadata
pull origin 
From github.com:neuropoly/data-multi-subject-test
 * branch                git-annex  -> FETCH_HEAD
 * branch                synced/git-annex -> FETCH_HEAD
ok
push origin 
Enumerating objects: 36, done.
Counting objects: 100% (36/36), done.
Delta compression using up to 128 threads
Compressing objects: 100% (27/27), done.
Writing objects: 100% (30/30), 2.61 KiB | 177.00 KiB/s, done.
Total 30 (delta 15), reused 1 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (15/15), completed with 4 local objects.
To github.com:neuropoly/data-multi-subject-test.git
   3145681b3..8d4222dd0  git-annex -> synced/git-annex
ok
p115628@joplin:~/data-multi-subject-test$ git push
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 128 threads
Compressing objects: 100% (3/3), done.
Writing objects: 100% (3/3), 360 bytes | 360.00 KiB/s, done.
Total 3 (delta 1), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (1/1), completed with 1 local object.
To github.com:neuropoly/data-multi-subject-test.git
   11a1202ad..77e80fcbd  master -> master
```

</details>

Now I've got a second update in the upstream, and my downstream is out of sync:

![Screenshot 2022-08-24 at 02-37-33 kousu_data-multi-subject-test Multi-subject data for the Spine Generic project](https://user-images.githubusercontent.com/987487/186348062-5f445515-1ea5-494d-be8d-15bc0ad51052.png)

So, sync it up:

![Screenshot 2022-08-24 at 02-37-33 kousu_data-multi-subject-test Multi-subject data for the Spine Generic project](https://user-images.githubusercontent.com/987487/186348272-ccefa61b-fbdb-4d73-8e07-060cddc3e903.png)

But can I sync `git-annex` from the GUI as well?

![Screenshot 2022-08-24 at 02-38-25 kousu_data-multi-subject-test at git-annex](https://user-images.githubusercontent.com/987487/186348273-b86d9023-07b3-4440-9d58-6660417a9314.png)

..nope :x: 

### CLI

First I'm going to reset the state to where it was at the start of the CLI sync:

<details>

```
p115628@joplin:~/datasets/data-multi-subject-test$ git pull
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 3 (delta 1), reused 3 (delta 1), pack-reused 0
Unpacking objects: 100% (3/3), 340 bytes | 113.00 KiB/s, done.
From github.com:kousu/data-multi-subject-test
   11a1202ad..77e80fcbd  master     -> origin/master
Updating 11a1202ad..77e80fcbd
Fast-forward
 bomp.nii.gz | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
p115628@joplin:~/datasets/data-multi-subject-test$ git log
commit 77e80fcbd23d2fc399c7e51b5d873d40ef8e21ed (HEAD -> master, origin/master, origin/HEAD)
Author: Nick Guenther <nick.guenther@polymtl.ca>
Date:   Wed Aug 24 02:33:52 2022 -0400

    Update bomp

commit 11a1202ad00f9e8459d74bbc86aec3b7769497bd
Author: Nick Guenther <nick.guenther@polymtl.ca>
Date:   Wed Aug 24 01:51:39 2022 -0400

    Make an update

commit 42e58865e3f26266d6b9899518942e317c85721f
Author: mbondy023 <98548430+mbondy023@users.noreply.github.com>
Date:   Wed Jul 20 14:12:08 2022 -0500

    add brain_t1 category into exclude.yml file (#127)

commit 082dd0d7b5ad1eedf53ea9765910e2d68769d355
Author: Alexandru Foias <alexfoias@users.noreply.github.com>
Date:   Tue Apr 12 10:50:06 2022 -0400

    Fix defaced images (#119)
    
    * add sub-balgrist01_T1w
    
    * add sub-balgrist02_T1w
    
    * add fixe defaced images
    
    * fix missing defaced
    
    * fix T1w images for cmrra and cmrrb
p115628@joplin:~/datasets/data-multi-subject-test$ git reset --hard HEAD~
HEAD is now at 11a1202ad Make an update
p115628@joplin:~/datasets/data-multi-subject-test$ git push -f
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
To github.com:kousu/data-multi-subject-test.git
 + 77e80fcbd...11a1202ad master -> master (forced update)
p115628@joplin:~/datasets/data-multi-subject-test$ git remote -v
amazon
origin  git@github.com:kousu/data-multi-subject-test.git (fetch)
origin  git@github.com:kousu/data-multi-subject-test.git (push)
```

</details>

Now what about this?

````suggestion
   If `spine-generic/data-multi-subject` is missing, you can add it with:
   ```
   git remote add upstream https://github.com/spine-generic/data-multi-subject.git
   ```

2. Then, to update your local clone, make sure to fetch the `git-annex` branch from `spine-generic/data-multi-subject` before running `git annex get`:
   ```
   git pull upstream master && annex get .
   ```
````

<details><summary>Test</summary>

```
p115628@joplin:~/datasets/data-multi-subject-test$ git pull upstream master
From https://github.com/neuropoly/data-multi-subject-test
 * branch                master     -> FETCH_HEAD
 * [new branch]          master     -> upstream/master
Updating 11a1202ad..77e80fcbd
git-annex: git status will show bomp.nii.gz to be modified, since content availability has changed and git-annex was unable to update the index. This is only a cosmetic problem affecting git status; git add, git commit, etc won't be affected. To fix the git status display, you can run: git update-index -q --refresh bomp.nii.gz
Fast-forward
 bomp.nii.gz | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
p115628@joplin:~/datasets/data-multi-subject-test$ git annex get bomp.nii.gz
get bomp.nii.gz 
  Remote upstream not usable by git-annex; setting annex-ignore
(not available) 
  No other repository is known to contain the file.

  (Note that these git remotes have annex-ignore set: origin upstream)
failed
git-annex: get: 1 failed
```

</details>

Okay not that. `pull remote branch` pulls *only* that branch, whereas `pull` pulls ALL branches. So :x: 

### CLI variant

````suggestion
   If `spine-generic/data-multi-subject` is missing, you can add it with:
   ```
   git remote add upstream https://github.com/spine-generic/data-multi-subject.git
   ```

2. Then, to update your local clone, make sure to fetch the `git-annex` branch from `spine-generic/data-multi-subject` before running `git annex get`:
   ```
   git fetch upstream && git pull upstream master && git annex get .
   ```
````

<details><summary>Test</summary>

```
p115628@joplin:~/datasets$ chmod -R +w data-multi-subject-test/
p115628@joplin:~/datasets$ rm -rf data-multi-subject-test/
p115628@joplin:~/datasets$ git clone git@github.com:kousu/data-multi-subject-test.git
Cloning into 'data-multi-subject-test'...
remote: Enumerating objects: 125584, done.
remote: Counting objects: 100% (29027/29027), done.
remote: Compressing objects: 100% (21176/21176), done.
remote: Total 125584 (delta 6415), reused 27989 (delta 6355), pack-reused 96557
Receiving objects: 100% (125584/125584), 15.76 MiB | 13.12 MiB/s, done.
Resolving deltas: 100% (46628/46628), done.
p115628@joplin:~/datasets$ cd data-multi-subject-test/
p115628@joplin:~/datasets/data-multi-subject-test$ git remote add upstream https://github.com/neuropoly/data-multi-subject-test.git
p115628@joplin:~/datasets/data-multi-subject-test$ git fetch upstream && git pull upstream master && git annex get bomp.nii.gz 
remote: Enumerating objects: 32, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (14/14), done.
remote: Total 32 (delta 16), reused 32 (delta 16), pack-reused 0
Unpacking objects: 100% (32/32), 2.90 KiB | 69.00 KiB/s, done.
From https://github.com/neuropoly/data-multi-subject-test
 * [new branch]          af/fix_images                        -> upstream/af/fix_images
 * [new branch]          af/update_participants_tsv           -> upstream/af/update_participants_tsv
 * [new branch]          af/update_sub-ucl06_GH_direct        -> upstream/af/update_sub-ucl06_GH_direct
 * [new branch]          git-annex                            -> upstream/git-annex
 * [new branch]          master                               -> upstream/master
 * [new branch]          mguaypaq/work-from-fork              -> upstream/mguaypaq/work-from-fork
 * [new branch]          mt-entity                            -> upstream/mt-entity
 * [new branch]          ng/ci-cache                          -> upstream/ng/ci-cache
 * [new branch]          renelabounek-master                  -> upstream/renelabounek-master
 * [new branch]          rl/height-weight                     -> upstream/rl/height-weight
 * [new branch]          sb/121-move-derivatives              -> upstream/sb/121-move-derivatives
 * [new branch]          synced/af/csfseg_json                -> upstream/synced/af/csfseg_json
 * [new branch]          synced/git-annex                     -> upstream/synced/git-annex
 * [new branch]          synced/jca/17-defacing               -> upstream/synced/jca/17-defacing
 * [new branch]          synced/jca/21-travis                 -> upstream/synced/jca/21-travis
 * [new branch]          synced/jca/7-check-data-consistency  -> upstream/synced/jca/7-check-data-consistency
 * [new branch]          synced/jca/derivatives-r20200830-mts -> upstream/synced/jca/derivatives-r20200830-mts
 * [new branch]          synced/jca/participants              -> upstream/synced/jca/participants
 * [new branch]          synced/jca/readme                    -> upstream/synced/jca/readme
 * [new branch]          synced/jca/reupload-disc-labels      -> upstream/synced/jca/reupload-disc-labels
 * [new branch]          synced/jca/update-derivatives        -> upstream/synced/jca/update-derivatives
From https://github.com/neuropoly/data-multi-subject-test
 * branch                master     -> FETCH_HEAD
Updating 11a1202ad..77e80fcbd
Fast-forward
 bomp.nii.gz | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
(merging origin/git-annex upstream/git-annex into git-annex...)
(recording state in git...)
(scanning for unlocked files...)
get bomp.nii.gz (from amazon...) 
(checksum...) ok                    
(recording state in git...)
```

</details>


This form should work whether or not someone combines it with the GitHub GUI :heavy_check_mark: